java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -3922,7 +3922,7 @@ public Table aggregateWindowsOverRanges(AggregationOverWindow... windowAggregate
             break;
           default:
             throw new IllegalArgumentException("Expected range-based window orderBy's " +
-                "type: integral (Boolean-exclusive) and timestamp");
+                "type: integral (Boolean-exclusive), decimal, and timestamp");
```

Comment:
I wonder if entire of this `switch` can be replaced by:
```
static Set<...> supportedTypes = Set<...>(INT8, INT16, ......);
if(!supportedTypes.contains(orderByType.getTypeId())) {
  throw new Exception(...);
}
```

Reply:
> I wonder if entire of this `switch` can be replaced by:

academically yes. but what is the benefit? Is it more readable?  still all constants  are listed.

The runtime code is more expensive because instead of a compact switch table, an array of offsets to jump to, there will be a full-fledged HashMap with boxing/unboxing

Reply:
Yeah, I'm afraid I concur with @gerashegalov's assessment. I'm not sure that's necessarily cleaner.